src/components/checkbox/checkbox.ts

```diff
@@ -248,6 +215,35 @@ export class MdCheckbox implements ControlValueAccessor {
     }
   }
 
+  /*
+   * We use a hidden native input field to handle changes to focus state via keyboard navigation,
+   * with visual rendering done separately. The native element is kept in sync with the overall
+   * state of the component.
+   */
+  onInputFocus() {
+    this.hasFocus = true;
+  }
+
+  onInputBlur() {
+    this.hasFocus = false;
```

Comment:
Should this set the ValueAccessor to touched as well? Excuse me if I'am wrong. Just wondering, because the `(blur)` host binding still exists on the components root.